motionplan/motionPlanner.go

```diff
@@ -566,10 +575,10 @@ func CheckPlan(
 			} else {
 				poseInPath = spatialmath.Compose(poseInPath, errorState)
 			}
-
 			modifiedSegment := &ik.State{Frame: sf, Position: poseInPath}
+			// check the state of the robot for collision
 			if isValid, _ := sfPlanner.planOpts.CheckStateConstraints(modifiedSegment); !isValid {
-				return fmt.Errorf("found collsion between positions %v and %v", currentPose.Point(), nextPose.Point())
+				return fmt.Errorf("found collsion between positions %v and %v", currentPose.Point(), poseInPath.Point())
```

Comment:
This is not the correct error message.

Either this should be "between `currentPose` and `nextPose`", or "at `poseInPath`".